framework/src/utils/RankThreeTensor.C

```diff
@@ -149,34 +136,27 @@ RankThreeTensor
 RankThreeTensor::operator/(const Real b) const
 {
   RankThreeTensor result;
-  const RankThreeTensor & a = *this;
 
-  for (unsigned int i = 0; i < N; ++i)
-    for (unsigned int j = 0; j < N; ++j)
-      for (unsigned int k = 0; k < N; ++k)
-        result(i, j, k) = a(i, j, k) / b;
+  for (unsigned int i = 0; i < N3; ++i)
+    result._vals[i] = _vals[i] / b;
 
   return result;
 }
 
 RankThreeTensor &
 RankThreeTensor::operator/=(const Real a)
 {
-  for (unsigned int i = 0; i < N; ++i)
-    for (unsigned int j = 0; j < N; ++j)
-      for (unsigned int k = 0; k < N; ++k)
-        _vals[i][j][k] /= a;
+  for (unsigned int i = 0; i < N3; ++i)
+    _vals[i] /= a;
 
   return *this;
 }
 
 RankThreeTensor &
 RankThreeTensor::operator+=(const RankThreeTensor & a)
 {
-  for (unsigned int i = 0; i < N; ++i)
-    for (unsigned int j = 0; j < N; ++j)
-      for (unsigned int k = 0; k < N; ++k)
-        _vals[i][j][k] += a(i, j, k);
+  for (unsigned int i = 0; i < N3; ++i)
+    _vals[i] += a._vals[i];
```

Comment:
http://stackoverflow.com/questions/4707012/is-it-better-to-use-stdmemcpy-or-stdcopy-in-terms-to-performance